chain/owasm/resolver.go

```diff
@@ -123,25 +130,40 @@ func (r *resolver) resolveRequestExternalData(vm *exec.VirtualMachine) int64 {
 	return 0
 }
 
+func (r *resolver) getExternalDataFromCache(externalDataID int64, validatorIndex int64) ([]byte, error) {
```

Comment:
This won't work if it's called two times and error. The first call will return error but the second call won't.